central/complianceoperator/v2/scanconfigurations/service/convert.go

```diff
@@ -409,6 +411,20 @@ func convertStorageSnapshotToV2Snapshot(ctx context.Context, snapshot *storage.C
 	if err != nil {
 		return nil, errors.Wrap(err, "Unable to convert ScanConfiguration to ScanStatus")
 	}
+	isDownloadReady := false
+	if snapshot.GetReportStatus().GetReportNotificationMethod() == storage.ComplianceOperatorReportStatus_DOWNLOAD {
+		if snapshot.GetReportStatus().GetRunState() == storage.ComplianceOperatorReportStatus_GENERATED ||
+			snapshot.GetReportStatus().GetRunState() == storage.ComplianceOperatorReportStatus_DELIVERED {
```

Comment:
This condition is pretty hard to read despite being already split into two `if`s. I suggest exporting it into function

```go
func shallCheckDownload(reportStatus *storage.ReportStatus) bool {
	return reportStatus.GetReportNotificationMethod() == storage.ComplianceOperatorReportStatus_DOWNLOAD && 
		(reportStatus.GetRunState() == storage.ComplianceOperatorReportStatus_GENERATED || reportStatus.GetRunState() == storage.ComplianceOperatorReportStatus_DELIVERED)
}
```

Reply:
Done